src/main.rs

```diff
@@ -200,7 +196,7 @@ async fn handle_request(
             std::path::PathBuf::from(uri.path())
                 .clean()
                 .to_str()
-                .ok_or(anyhow!("failed to clean path: {:?}", uri.path()))?,
+                .ok_or_else(|| anyhow!("failed to clean path: {:?}", uri.path()))?,
```

Comment:
I think you can use anyhow's `Context` on options: (same for the other `ok_or_else`s)
```suggestion
                .with_context(|| format!("failed to clean path: {:?}", uri.path()))?,
```

Reply:
haha, i had even used this pattern in other places u_u